assets/js/api.js

```diff
@@ -1,644 +1,368 @@
-  	  $(permitsJson.features).each(function(key, data) {
-
-        if (dataType == "permits") {
-          var date = data.properties.issuedate;
-        } else {
-          var date = data.properties.demolition_date;
-        }
-        var propStartYear = date;//moment(date);//.get('year');
-        var propEndYear = end;//moment(date).add(150, 'days');//.get('year');
-        data.properties['start'] = propStartYear;
-        data.properties['end'] = propEndYear;
-
-      });
-      // permitsLayer.addLayer(permitsLayer[key]);
-      timelineLayer.addData(permitsJson);
-      console.log(this.url);
-      // permitsLayer.addTo(map);
-    })
-    .fail(function() {
-      console.log("Failed to fetch permits json data");
-    });
-    // promises.push(promise);
-  }
-
-  // Get crime data
-  // store in Taffy db
-  var crimeUrl = "http://ec2-52-88-193-136.us-west-2.compute.amazonaws.com/services/crimes.json";
-
-  function getCrimesYear(nbhood, yearRange) {
-    var crimeCount = 0;
-    var crimesByYear = {}; // need year to count mapping to sync count animation to timeline
-    var crimesPromises = [];
-    for (var i = 0; i < yearRange.length; i++) {
-      var promise = $.ajax({
-    	  method: "GET",
-    	  url: crimeUrl,
-    	  data: {
-          query: "perNeighborhoodPerYear",
-          year: yearRange[i],
-          // hardcoding violent crime parameter for the time being:
-          type: "violent"
-    	  }
-    	})
-      .done(function(data) {
-    	  var crimesJson = data;
-        crimesTaffyList = [];
-        var crimesInNbhood = 0;
-        for (var j = 0; j < (crimesJson.rows).length; j++) {
-          var rec = {
-            year: yearRange[i],
-            name: crimesJson['rows'][j][0],
-            num: crimesJson['rows'][j][1]
-          };
-          crimesTaffyList.push(rec);
-
-          if (crimesJson['rows'][j][0] == nbhood) {
-            crimesInNbhood += crimesJson['rows'][j][1];
-          }
-
-        }
-        if (typeof crimesDb == "undefined") {
-          crimesDb = TAFFY(crimesTaffyList);
-        } else {
-          crimesDb.insert(crimesTaffyList);
-        }
-        crimeCount += crimesInNbhood;
-        //$('#crimetotal').append(crimesInNbhood + '<br>');
-
-      })
-      .fail(function() {
-        console.log("Failed to fetch crimes.");
-      });
-      crimesPromises.push(promise);
+    nbhoodDb = TAFFY(nbhoodTaffyList);
+  })
+  .fail(function() {
+    console.log("Failed to fetch neighborhood list.");
+  });
+}
+
+function getNbhoodShape(nbhood) {
+  $.ajax({
+    method: "GET",
+    url: nbhoodShapesUrl,
+    data: {
+      city: "portland",
+      name: nbhood
     }
-
-    $.when.apply($, crimesPromises).done(function() {
-      $('#crimetotal').append(crimeCount);
-      nbhoodLayer.setStyle({
-        fillColor: getColor(crimeCount)
-      });
-    }).fail(function() {
-      console.log('Failed to finish iterating over crime.');
+  })
+  .done(function(data) {
```

Comment:
I'm not sure if nbhoodLayer should be in the presentation level of code or the data level? Since it has style, maybe this work should be taking place in the presentation code?